arangod/Replication2/ReplicatedLog/LogLeader.h

```diff
@@ -146,6 +147,8 @@ class LogLeader : public std::enable_shared_from_this<LogLeader>,
 
   auto waitForLeadership() -> WaitForFuture override;
 
+  auto waitForResign() -> futures::Future<futures::Unit> override;
```

Comment:
I think we should annotate all the `waitFor` functions, that return a future, with ``[[nodiscard]]`. Otherwise one might assume that the call is blocking. (Has happened before)

Reply:
True!